Assets/Scripts/Game/WeaponManager.cs

```diff
@@ -413,49 +414,43 @@ public void Reset()
+        public bool WeaponEnvDamage(DaggerfallUnityItem strikingWeapon, RaycastHit hit)
         {
-            DaggerfallUnityItem strikingWeapon = usingRightHand ? currentRightHandWeapon : currentLeftHandWeapon;
-
-            if (arrowHit)
+            // Check if hit has an DaggerfallAction component
+            DaggerfallAction action = hit.transform.gameObject.GetComponent<DaggerfallAction>();
+            if (action)
             {
-                strikingWeapon = lastBowUsed;
+                action.Receive(player, DaggerfallAction.TriggerTypes.Attack);
             }
-            else
-            {
-                // Check if hit has an DaggerfallAction component
-                DaggerfallAction action = hit.transform.gameObject.GetComponent<DaggerfallAction>();
-                if (action)
-                {
-                    action.Receive(player, DaggerfallAction.TriggerTypes.Attack);
-                }
 
-                // Check if hit has an DaggerfallActionDoor component
-                DaggerfallActionDoor actionDoor = hit.transform.gameObject.GetComponent<DaggerfallActionDoor>();
-                if (actionDoor)
-                {
-                    actionDoor.AttemptBash(true);
-                    return false;
-                }
+            // Check if hit has an DaggerfallActionDoor component
+            DaggerfallActionDoor actionDoor = hit.transform.gameObject.GetComponent<DaggerfallActionDoor>();
+            if (actionDoor)
+            {
+                actionDoor.AttemptBash(true);
+                return true;
+            }
 
-                // Check if player hit a static exterior door
-                if (GameManager.Instance.PlayerActivate.AttemptExteriorDoorBash(hit))
-                {
-                    return false;
-                }
+            // Check if player hit a static exterior door
+            if (GameManager.Instance.PlayerActivate.AttemptExteriorDoorBash(hit))
+            {
+                return true;
+            }
 
-                // Make hitting walls do a thud or clinging sound (not in classic)
-                if (GameObjectHelper.IsStaticGeometry(hit.transform.gameObject))
-                {
-                    DaggerfallUI.Instance.PlayOneShot(strikingWeapon == null ? SoundClips.Hit2 : SoundClips.Parry6);
-                    return false;
-                }
+            // Make hitting walls do a thud or clinging sound (not in classic)
+            if (GameObjectHelper.IsStaticGeometry(hit.transform.gameObject))
+            {
+                DaggerfallUI.Instance.PlayOneShot(strikingWeapon == null ? SoundClips.Hit2 : SoundClips.Parry6);
+                return true;
             }
 
-            // Set up for use below
-            Transform hitTransform = arrowHit ? arrowHitCollider.gameObject.transform : hit.transform;
-            Vector3 impactPosition = arrowHit ? hitTransform.position : hit.point;
+            return false;
```

Comment:
There's a small issue with non-static environment, say portcullis or blue bars, that prevent you from hitting enemies thru: WeaponEnvDamage returns false so RayHit fallback does not happen.
Not sure how to improve test here to take into account non-static environment (or mark portcullis and blue bars as static env, if that makes sense)